hybrasyl/Monolith.cs

```diff
@@ -133,52 +136,171 @@ private static void SpawnMonster(Monster monster, Map map)
             //Game.World.Maps[mapId].InsertCreature(monster);
             //Logger.DebugFormat("Spawning monster: {0} at {1}, {2}", monster.Name, (int) monster.X, (int) monster.Y);
         }
+    }
+
+    internal class MonolithControl
+    {
+        public static readonly ILog Logger = LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
+
+        private IEnumerable<Map> _maps => Game.World.WorldData.Values<Map>();
+        private static Random _random;
+
+        internal MonolithControl()
+        {
+            _random = new Random();
+        }
+        
+        public void Start()
+        {
+            while (true)
+            {
+                foreach (var map in _maps)
+                {
+                    //var t = Task.Run(() =>
+                    //{
+                        foreach (var mob in map.EntityTree.GetAllObjects().OfType<Monster>())
+                        {
+                            Evaluate(mob, map);
+                        }
```

Comment:
Realistically, given what `Evaluate` does, this should be handled using a `HybrasylControlMessage` type, e.g. `ControlMessage_EntityEvaluate` or something. That should then have a defined handler in World.cs (   `SetControlMessageHandlers`) which does all of this (so that the logic runs in the message consumption thread).

Running game logic, especially that which modifies state, outside of the consumer thread, is asking for nightmares. We make no guarantee of thread safety across most of the project because the assumption is that only the consumer thread is (directly) modifying game state.

Reply:
I agree with this, hence the empty method below.